### PR TITLE
Subtitle tests and type controls

### LINKED ISSUE #269: [Bug]: First line of multiline captions are ignored by the library
### What happened?

Subtitles with multiple lines are not parsed correctly. The first line is always ignored. This only occurs on files with LF line ending.
When converted to CRLF it works fine
Example:
Input
```
WEBVTT

00:00:00.000 --> 00:00:10.000
- Test 1\n- Test 2\n- Test 3

00:00:10.000 --> 00:00:20.000
- Test 1
- Test 2
- Test 3
```
Output:
```json
[{
    "type": "caption",
    "index": 1,
    "start": 0,
    "end": 10000,
    "duration": 10000,
    "content": "- Test 2 - Test 3",
    "text": "- Test 2 - Test 3"
 },
{
    "type": "caption",
    "index": 2,
    "start": 10000,
    "end": 20000,
    "duration": 10000,
    "content": "- Test 2 - Test 3",
    "text": "- Test 2 - Test 3"
 }]
```

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### Steps to reproduce?

_No response_

### Other relevant information

~I have opened an issue in the library repo hopefully they'll look into it
https://github.com/leranjun/subsrt-ts/issues/2~
They looked into it

### PR DESCRIPTION
**Pull request by frost768**
- Filter non-subtitle files in providers
- Reset caption delay on every subtitle change and exiting video player. Delay should not persist between video sessions
- Add tests for subtitles
- ~Use forked version of subsrt-ts with the multiline support patch~ subsrt-ts updated (fix #269 )

### CODE REVIEW

**Comment by mrjvs** [comment]
looks good, however I don't see how this will fix #237.
It's still missing a track element?

**Comment by frost768** [comment]
> looks good, however I don't see how this will fix #237. It's still missing a track element?
it doesnt sorry, we can do it using pip events but it would be incompatible with firefox. there is no way to detect leaving the pip mode only entering it

**Comment by TheGloriousDuck** [comment]
> > looks good, however I don't see how this will fix #237. It's still missing a track element?
> 
> it doesnt sorry, we can do it using pip events but it will be incompatible with firefox. there is no way to detect leaving the pip mode on firefox only entering it

I'm looking here and it looks like it does?? I don't know, I could be wrong
https://developer.mozilla.org/en-US/docs/Web/API/Picture-in-Picture_API#events

**Comment by TheGloriousDuck** [comment]
I see. Could we still just implement it for browsers other than Firefox then? Better to have something than nothing

**Comment by frost768** [comment]
How can we clean the node_modules cache in the pipeline? It compares the official package with the forked one.

File: src/backend/providers/superstream/index.ts [hunk 1]
```diff
@@ -111,6 +115,30 @@ const getBestQuality = (list: any[]) => {
   );
 };
 
+const convertSubtitles = (subtitleGroup: any): MWCaption | null => {
+  let subtitles = subtitleGroup.subtitles;
+  subtitles = subtitles
+    .map((subFile: any) => {
```
Comment (JipFr) [inline src/backend/providers/superstream/index.ts:121]: Do we require explicit `any` / should these be typed?

File: src/backend/providers/superstream/index.ts [hunk 2]
```diff
@@ -111,6 +115,30 @@ const getBestQuality = (list: any[]) => {
...
+      if (!supported) return null;
+      const type = getMWCaptionTypeFromUrl(subFile.file_path);
+      return {
+        ...subFile,
+        type: type as MWCaptionType,
+      };
+    })
+    .filter(Boolean);
+
+  if (!subtitles.length) return null;
```
Comment (JipFr) [inline src/backend/providers/superstream/index.ts:132]: I feel like using a falsy check here feels a little weird, perhaps the following instead?
```suggestion
  if (subtitles.length === 0) return null;
```

File: src/utils/thumbnailCreator.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  const video = document.createElement("video");
+  video.src = videoUrl;
+  video.crossOrigin = "anonymous";
+
+  // Wait for the video metadata to load
+  const metadata = await new Promise((resolve, reject) => {
+    video.addEventListener("loadedmetadata", resolve);
+    video.addEventListener("error", reject);
+  });
+  console.log(metadata);
```
Comment (JipFr) [inline src/utils/thumbnailCreator.ts:14]: Console.log should probably be removed
```suggestion
```

File: src/video/components/actions/CaptionRendererAction.tsx [hunk 4]
```diff
@@ -63,20 +63,37 @@ export function CaptionRendererAction({
...
     } else {
       captions.current = [];
     }
   }, [source?.caption?.url]);
-
+  useEffect(() => {
+    // reset delay after video ends
+    return () => setCaptionDelay(0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (JipFr) [inline src/video/components/actions/CaptionRendererAction.tsx:76]: Why is exhaustive deps disabled? What iss the problem with adding `setCaptionDelay` as a dep?

Comment (frost768) [reply src/video/components/actions/CaptionRendererAction.tsx:76]: it causes too much rendering. not sure why

Comment (mrjvs) [reply src/video/components/actions/CaptionRendererAction.tsx:76]: I wouldn't make a unmount to reset, I'd rather set it to 0 on video player load and when source url changes.

File: src/utils/thumbnailCreator.ts (hunk 3 above)
Comment (frost768) [reply src/utils/thumbnailCreator.ts:14]: i think the whole file should be removed. it doesn't have anything to with subtitles and is not complete as a feature

File: src/video/components/actions/CaptionRendererAction.tsx (hunk 4 above)
Comment (frost768) [reply src/video/components/actions/CaptionRendererAction.tsx:76]: So should I put the useEffect it in `VideoPlayer.tsx`?

File: src/backend/providers/superstream/index.ts (hunk 1 above)
Comment (frost768) [reply src/backend/providers/superstream/index.ts:121]: As @mrjvs said in discord the providers are prone to change so I think it should stay as any

File: src/utils/thumbnailCreator.ts (hunk 3 above)
Comment (JipFr) [reply src/utils/thumbnailCreator.ts:14]: Jvs and I agree — remove it. We’re excited to see it in the app though!

File: src/video/components/actions/CaptionRendererAction.tsx (hunk 4 above)
Comment (cruizm-cyber) [reply src/video/components/actions/CaptionRendererAction.tsx:76]: hey ladys

**Review by mrjvs — APPROVED** [review]

**Pull request merged**